packages/protocol/contracts/bridge/libs/LibBridgeProcess.sol

```diff
@@ -105,13 +111,17 @@ library LibBridgeProcess {
                 ? gasleft()
                 : message.gasLimit;
 
-            // this will call receiveERC20 on the tokenVault, sending the tokens to the user
-            bool success = LibBridgeInvoke.invokeMessageCall({
-                state: state,
-                message: message,
-                msgHash: msgHash,
-                gasLimit: gasLimit
-            });
+            bool success = true;
+
+            if (message.data.length != 0) {
+                // this will call receiveERC20 on the tokenVault, sending the tokens to the user
+                success = LibBridgeInvoke.invokeMessageCall({
+                    state: state,
+                    message: message,
+                    msgHash: msgHash,
+                    gasLimit: gasLimit
+                });
+            }
 
             if (success) {
                 status = LibBridgeStatus.MessageStatus.DONE;
```

Comment:
The current `bool success= true ` is basically serving the 'else' clause of the `message.data.length != 0`.
If we rearrange it, we save some gas on variable initialization.
```

            // use the specified message gas limit if not called by the owner
            uint256 gasLimit = msg.sender == message.owner ? gasleft() : message.gasLimit;

            if (message.data.length != 0) {
                // this will call receiveERC20 on the tokenVault, sending the tokens to the user
                bool success = LibBridgeInvoke.invokeMessageCall({
                    state: state,
                    message: message,
                    msgHash: msgHash,
                    gasLimit: gasLimit
                });

                if (success) {
                    status = LibBridgeStatus.MessageStatus.DONE;
                } else {
                    status = LibBridgeStatus.MessageStatus.RETRIABLE;
                    if (ethVault != address(0)) {
                        ethVault.sendEther(message.callValue);
                    }
                }
            } else {
                status = LibBridgeStatus.MessageStatus.DONE;
            }
```

Reply:
Daniel said to remove the length check so I did